src/Data/Aeson/Types/FromJSON.hs

```diff
@@ -1517,6 +1518,7 @@ instance FromJSON1 Maybe where
 
 instance (FromJSON a) => FromJSON (Maybe a) where
     parseJSON = parseJSON1
+    omittedDefault = Just Nothing
```

Comment:
A field absent in JSON will be populated by `Nothing` in the Haskell representation, preserving the current behavior of `FromJSON` for `Maybe a` fields, without needing overlapping instances.